xarray/core/utils.py

```diff
@@ -371,6 +371,35 @@ def ordered_dict_intersection(
     return new_dict
 
 
+def ordered_dict_union(
+    first_dict: MutableMapping[K, V],
```

Comment:
Why does this need to be mutable?

Reply:
`update_safety_check()` expects a `MutableMapping` as its first argument. I can't see why that needs to be mutable though... Should I change both `update_safety_check()` and `ordered_dict_union()` to have their first arg be `Mapping[K, V]`?

Reply:
I think so, yes

Reply:
Should this method be called `ordered`?

Reply:
Not sure... If the comparison defined by the `compat` argument means the dict values are identical, then the order of first and second argument doesn't matter, but if something is passed as the `compat` argument that returns `True` when they are 'compatible' but not identical, then the value from the first dict will be used in the result. I think that's the meaning of 'ordered' here (I copied the naming from `ordered_dict_intersection` just above).

Reply:
Ah I reckon the above method is left over from when we used `OrderedDict`s. Shall we rename this to something like `compat_dict_intersection` (or whatever you think is better)? 

Could also add a TODO in the above method (but let's change it in a different PR, no need to slow down this)